lib/tokenize.cpp

```diff
@@ -8349,24 +8349,30 @@ void Tokenizer::findGarbageCode() const
     }
 
     // case keyword must be inside switch
+    bool inSwitchStmt = false;
+    const Token *switchStmtEnd = 0;
     for (const Token *tok = tokens(); tok; tok = tok->next()) {
         if (Token::simpleMatch(tok, "switch (")) {
+            inSwitchStmt = true;
             if (Token::simpleMatch(tok->linkAt(1), ") {")) {
                 tok = tok->linkAt(1)->linkAt(1);
+                inSwitchStmt = false;
+                continue;
```

Comment:
If you have `continue` then you don't need a `else`. I would suggest that the `continue` is kept and the `else` removed.